app/src/test/java/nl/tudelft/b_b_w/controller/ApiUnitTest.java

```diff
@@ -0,0 +1,150 @@
+package nl.tudelft.b_b_w.controller;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.RuntimeEnvironment;
+import org.robolectric.annotation.Config;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import nl.tudelft.b_b_w.BuildConfig;
+import nl.tudelft.b_b_w.model.BlockFactory;
+import nl.tudelft.b_b_w.model.User;
+
+import static junit.framework.Assert.assertEquals;
+import static junit.framework.Assert.assertNotSame;
+import static junit.framework.Assert.assertTrue;
+import static nl.tudelft.b_b_w.controller.Api.createUser;
+
+/**
+ * Test the API.
+ * TODO revoke does not work yet, Luat is working on it. Also extra unittests.
+ */
+@RunWith(RobolectricTestRunner.class)
+@Config(constants = BuildConfig.class,sdk= 21,  manifest = "src/main/AndroidManifest.xml")
+public class ApiUnitTest {
+    private BlockController bc;
+    private User userA, userB, userC, userD, userE;
+
+    /**
+     * User A has one public key, user B two
+     * The blockchain of each user:
+     * A: add pkA
+     * B: add pkA
+     * C: empty
+     * D: add pkA, revoke pkA
+     * E: add pkE, revoke pkE
+     * F: add pkF1, add pkF2
+     * G: add pkF1, add pkF2, add pkG
+     * All is tested from viewpoint of A.
+     */
+    @Before
+    public void setUp() {
+        bc = new BlockController(RuntimeEnvironment.application);
+        Api.init(RuntimeEnvironment.application);
+
+        userA = createUser("Antro", "NL81INGB0000000000");
+        userB = createUser("Besse", "NL81INGB0000000000");
+        userC = createUser("Cacao", "NL81INGB0000000000");
+        userD = createUser("Draffyg", "NL81INGB0000000000");
+        userE = createUser("Erwti", "NL81INGB0000000000");
+
+        // public static Block getBlock(String type, String _owner, String _ownHash, String _previousHashChain, String _previousHashSender, String _publicKey, String _iban, int _trustValue) throws IllegalArgumentException {
+        // public static Block getBlock(User owner, User user, String ownHash, String previousHashChain, String previousHashSender, String publicKey, boolean revoke) {
+
+        // A: add A
+        bc.addBlock(BlockFactory.getBlock("BLOCK", userA.getName(), "roothashA", "prevhashchain", "root", "pkroot", "NL81...", 0));
```

Comment:
Why didn't you get the iban from the user class and define all attributes once and use the attributes then?